### ISSUE TITLE
fixed skip writing a message if an interrupt is received

### ISSUE THREAD

**Issue by RobAustin** [OpenHFT/Chronicle-Queue#745]
ChronicleQueueBuilder.checkInterrupts == true will cause the appender to skip writing a message if an interrupt is received. The ChronicleQueueBuilder.checkInterrupts==true by default. This change sets it to false by default.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.20.105](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.20.105), [BOM-2.20.144](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.144)

**Comment by JerryShea** [comment]
Looking at this again, the code rolls back if an interrupt is received
```
                final boolean interrupted = checkInterrupts && Thread.currentThread().isInterrupted();
                if (rollbackOnClose || interrupted) {
                    doRollback(interrupted);
                    return;
                }
```
and is working on the assumption that its thread is about to die. To capture the original intent, I think better for us to throw an exception if we detect we have been interrupted e.g.
```
                final boolean interrupted = checkInterrupts && Thread.currentThread().isInterrupted();
                if (interrupted)
                    throw new InterruptedException();
                if (rollbackOnClose) {
                    doRollback(interrupted);
                    return;
                }
```
